src/compiler/tracing.ts

```diff
@@ -131,9 +131,9 @@ namespace ts { // eslint-disable-line one-namespace-per-file
             }
             eventStack.push({ phase, name, args, time: 1000 * timestamp(), separateBeginAndEnd });
         }
-        export function pop() {
+        export function pop(results?: Args) {
             Debug.assert(eventStack.length > 0);
-            writeStackEvent(eventStack.length - 1, 1000 * timestamp());
+            writeStackEvent(eventStack.length - 1, 1000 * timestamp(), results);

```

Comment:
Should this throw if `separateBeginAndEnd`?

Reply:
I didn’t check whether it works on `E` events, but I can take your word that it doesn’t and do that.